### PR TITLE
Update to and fix latest lints

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: analysis_options.yaml [hunk 1]
```diff
@@ -1,6 +1,5 @@
...
-linter:
-  rules:
-    - avoid_dynamic_calls
-    - directives_ordering
-    - unawaited_futures
+include: package:dart_flutter_team_lints/analysis_options.yaml
+
+analyzer:
+  errors:
+    comment_references: ignore
```
Comment (kevmoo) [inline analysis_options.yaml:5]: LOTS of failures here!

File: lib/src/line_splitting/solve_state_queue.dart [hunk 2]
```diff
@@ -14,7 +14,9 @@ import 'solve_state.dart';
 /// overflow characters. When a new state is added to the heap, it will be
 /// discarded, or a previously enqueued one will be discarded, if two overlap.
 class SolveStateQueue {
-  /// Initial capacity of a queue when created, or when added to after a [clear].
+  /// Initial capacity of a queue when created, or when added to after a
+  /// `clear`.
```
Comment (kevmoo) [inline lib/src/line_splitting/solve_state_queue.dart:18]: Zero idea where "clear" went.

File: analysis_options.yaml [hunk 3]
```diff
@@ -1,6 +1,5 @@
-include: package:lints/recommended.yaml
-linter:
-  rules:
-    - avoid_dynamic_calls
```
Comment (kevmoo) [inline analysis_options.yaml:4]: all of these are in `dart_flutter_team_lints`

File: lib/src/front_end/comment_writer.dart [hunk 4]
```diff
@@ -106,6 +106,7 @@ mixin CommentWriter {
     // just override the script tag's line.
     if (token.previous!.type == TokenType.SCRIPT_TAG) previousLine = tokenLine;
 
+    // ignore: prefer_const_constructors
```
Comment (munificent) [inline lib/src/front_end/comment_writer.dart:109]: Does the linter really yell at you to use `const` here?! It's creating a mutable object. That's a bananas lint. I definitely don't want to enable it for dart_style.

File: lib/src/front_end/comment_writer.dart [hunk 5]
```diff
@@ -312,7 +313,7 @@ class CommentSequence extends ListBase<SourceComment> {
   SourceComment operator [](int index) => _comments[index];
 
   @override
-  operator []=(int index, SourceComment value) =>
+  void operator []=(int index, SourceComment value) =>
```
Comment (munificent) [inline lib/src/front_end/comment_writer.dart:316]: This return type is pointless. Index setters are like setters: they can never return a value anyway.

File: lib/src/front_end/delimited_list_builder.dart [hunk 6]
```diff
@@ -3,8 +3,8 @@
 // BSD-style license that can be found in the LICENSE file.
 import 'package:analyzer/dart/ast/ast.dart';
 import 'package:analyzer/dart/ast/token.dart';
-import 'package:dart_style/src/ast_extensions.dart';
```
Comment (munificent) [inline lib/src/front_end/delimited_list_builder.dart:6]: Nice catch. I wish VS Code didn't automatically use package imports when it auto-adds an import from the same package.

File: lib/src/line_splitting/solve_state_queue.dart (hunk 2 above)
Comment (munificent) [reply lib/src/line_splitting/solve_state_queue.dart:18]: This class was copied from HeapPriorityQueue and then heavily modified, so this comment is just referring to something that no longer applies to this class. You can just remove everything after "created".

**Review by munificent — CHANGES_REQUESTED** [review]

File: lib/src/front_end/comment_writer.dart (hunk 5 above)
Comment (kevmoo) [reply lib/src/front_end/comment_writer.dart:316]: Generally agree, so maybe the lint needs to be updated for this case!

File: lib/src/front_end/delimited_list_builder.dart (hunk 6 above)
Comment (kevmoo) [reply lib/src/front_end/delimited_list_builder.dart:6]: agreed!

File: lib/src/front_end/comment_writer.dart (hunk 4 above)
Comment (kevmoo) [reply lib/src/front_end/comment_writer.dart:109]: Yep!

If we don't want it enabled, there's a lot of churn to delete.

File: example/format.dart [hunk 7]
```diff
@@ -1,6 +1,9 @@
 // Copyright (c) 2015, the Dart project authors.  Please see the AUTHORS file
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
+
+// ignore_for_file: unreachable_from_main
```
[the remark this answers is not in the document]
Comment (kevmoo) [reply example/format.dart:5]: okay to just delete top-level functions that aren't used?

File: lib/src/line_splitting/solve_state_queue.dart (hunk 2 above)
Comment (kevmoo) [reply lib/src/line_splitting/solve_state_queue.dart:18]: done

File: lib/src/argument_list_visitor.dart [hunk 8]
```diff
@@ -96,10 +96,11 @@ class ArgumentListVisitor {
...
       this._arguments,
       this._functions,
-      this._argumentsAfterFunctions) {
-    assert(_functions == null || _argumentsAfterFunctions != null,
-        'If _functions is passed, _argumentsAfterFunctions must be too.');
-  }
+      this._argumentsAfterFunctions)
+      : assert(
+          _functions == null || _argumentsAfterFunctions != null,
+          'If _functions is passed, _argumentsAfterFunctions must be too.',
```
[the remark this answers is not in the document]
Comment (kevmoo) [reply lib/src/argument_list_visitor.dart:102]: done

File: lib/src/dart_formatter.dart [hunk 9]
```diff
@@ -82,8 +82,9 @@ class DartFormatter {
       throw ArgumentError('uri must be `null`, a Uri, or a String.');
     }
 
-    return formatSource(SourceCode(source, uri: uri, isCompilationUnit: true))
-        .text;
+    return formatSource(
+      SourceCode(source, uri: uri as String?, isCompilationUnit: true),
```
[the remark this answers is not in the document]
Comment (kevmoo) [reply lib/src/dart_formatter.dart:86]: done

File: lib/src/front_end/comment_writer.dart (hunk 5 above)
Comment (munificent) [reply lib/src/front_end/comment_writer.dart:316]: I believe the lint is doing the right thing. I enabled always_declare_return_types and tested it on:

```dart
class Foo {
  get getter => 1;
  set setter(int value) => 1;
  method() => 1;
  operator [](int index) => 1;
  operator []=(int index, int value) => 1;
}
```

I get lint warnings on `getter`, `method()`, and `operator []`, but not `setter` or `operator []=`. Maybe the fix is out of sync with the lint? Or maybe this was just a mistake fixing the lint? Either way, I think you can delete this `void` and it should still be lint clean.

**Review by munificent — APPROVED** [review]
This LGTM, but note that I have a very large PR out (#1336) that will probably collide with this in nasty ways and I really don't want that PR (which is blocking any other progress on the new back end) any harder to land than it already is.

Would you be OK with waiting until that's in and updating this PR with those changes first?

**Pull request merged**